Lastpass docs (#49)
helpfulengineering/devops: guidance/Credentials.md

# Helpful Engineering Shared Credentials

> Whenever possible we should create separate accounts for each user.  However in some instances it will be required to share a single set of credentials (ex. social media accounts, HE email accounts, etc).  If we have to share credentials with multiple people we will do so using a tool called Lastpass.  **Credentials should NEVER be shared in plaintext.  Please only share with lastpass so passwords remain encrypted at all times.**

Reviewer: ```suggestion
Whenever possible, no account should be shared with other users.  However, in some instances (e.g. social media accounts, some email accounts, *et cetera*) it will be required to share a single set of credentials with multiple users using a tool called [LastPass](https://www.lastpass.com).

 ⚠️ Credentials should **never** be shared outside LastPass: please **don't** send passwords in plaintext by any other means.
```

Bold and uppercase text can distract the reader: some bold words and a good warning emoji can be more effective.

> # Lastpass Master Account
> The devops and infosec teams maintains a master account where all credentials are stored.  Credentials are then shared from this account to other users after access has been approved.  When the infosec team rotates passwords regularly as a security best practice they will automatically update in other users lastpass vault.  

Reviewer: ```suggestion
## LastPass master account
The DevOps and Infosec teams maintain a master account where all credentials are stored and shared with the approved users. Passwords are being rotated regularly for security reasons, and the new ones will automatically synchronize in the other users' LastPass vaults.  
```

> # Creating a lastpass account

Reviewer: ```suggestion
## Creating a LastPass account
```

If you do not have a lastpass account please [create one](https://lastpass.com/create-account.php).  **DO NOT FORGET YOUR LASTPASS PASSWORD!**  It cannot be reset by the infosec team or lastpass.  Your password is used to generate your encryption key, so it is unique to you.  We also recommend installing the lastpass browser extension to your favorite browser.

After your account is created send the email address you used to sign up to the infosec team ([#skill-infosec](https://app.slack.com/client/TUTSYURT3/CV4TYGC1Z)) and they will share passwords with you.  If you are outside of the admin group in slack approval for access will be required.  The infosec team will reach out to the admin group for approval. 

[Lastpass sharing documentation](https://helpdesk.lastpass.com/sharing-4-0/)

> # Changing Passwords

Reviewer: ```suggestion
## Changing passwords
```

Lastpass can also [generate secure random passwords](https://helpdesk.lastpass.com/generating-a-password/).  Please use the following settings when generating passwords

 - length > 12
 - Uppercase letter
 - Lowercase letter
 - Numbers

>  - Symbols

Reviewer: ```suggestion
When changing a password, you should [generate the new one with LastPass](https://helpdesk.lastpass.com/generating-a-password/) (or equivalent) using the following settings:

* **All characters**:
  - [x] Uppercase
  - [x] Lowercase
  - [x] Numbers
  - [x] Symbols
* Length: **15 or more** characters.
```

Rising the password length to 15 or more characters as per [this recommendation](https://www.fbi.gov/contact-us/field-offices/portland/news/press-releases/oregon-fbi-tech-tuesday-building-a-digital-defense-with-passwords).<sup>1</sup> It seems a good idea, as passwords will be managed through LastPass and nobody will have to remember or type them; the former NIST's [8 character recommendation](https://pages.nist.gov/800-63-3/sp800-63b.html) was ludicrous for some hash formats.

<sup>1</sup> And not, `VoicesProtected2020WeAre` is not a secure password; better let a good entropy source choose it for you. [XKCD 936](https://xkcd.com/936/) is a good reference if (and only if) you rely in a trusted "standard" like BIP39.

> If you change a password please update the record in lastpass, this will share the new passwords to all other authorized users.

Reviewer: ```suggestion
Once you change a password, please update the record in LastPass, so it's automatically shared with all the authorized users.
```
